.typedoc/page-custom-logic.js

```diff
@@ -0,0 +1,170 @@
+const INTERNAL_MODULE_NAME = "internal";
+const GENERATED_DOC_RETURN_URL_LOCAL_STORAGE_KEY = "generatedDocReturnUrl";
+
+const isEmbeddingSdkApiDocsPage = (href) =>
+  href.includes("/embedding/sdk/api/");
+
+const storeReturnUrl = () => {
+  const ref = document.referrer;
+
+  const isRefFromNonApiPage = ref && !isEmbeddingSdkApiDocsPage(ref);
+
+  if (isRefFromNonApiPage) {
+    sessionStorage.setItem(GENERATED_DOC_RETURN_URL_LOCAL_STORAGE_KEY, ref);
+  }
+};
+
+/**
+ * Called from `navigationLinks` field of `typedoc.config.mjs
```

Comment:
It's great to have a comment like this especially, when a function is set in a global scope. Otherwise, it's hard to understand its necessity.